database/migrations/default/1742408937781_crash_ems_link/up.sql

```diff
@@ -0,0 +1,391 @@
+
+--
+-- add travel_mode column
+--
+alter table ems__incidents
+add column travel_mode text generated always as (
+    case
+        when lower(mvc_form_vehicle_type) = 'motorcycle' then 'Motorcycle'
+        when lower(mvc_form_vehicle_type) = 'moped' then 'Motorcycle'
+        when lower(pcr_cause_of_injury) like '%motorcy%' then 'Motorcycle'
+        when
+            lower(pcr_cause_of_injury) like '%scoot%'
+            or lower(pcr_patient_complaints) like '%scoot%'
+            then 'E-Scooter'
+        when
+            lower(pcr_cause_of_injury) like '%bike%'
+            or lower(pcr_cause_of_injury) like '%bicy%'
+            or lower(pcr_patient_complaints) like '%bike%' or lower(pcr_patient_complaints) like '%bicy%' then 'Bicycle'
+        when
+            lower(pcr_cause_of_injury) like '%pede%'
+            or lower(pcr_patient_complaints) like '%pede%'
+            or (lower(pcr_cause_of_injury) like '%auto%' and lower(pcr_cause_of_injury) like '%ped%')
+            or (lower(pcr_patient_complaints) like '%auto%' and lower(pcr_patient_complaints) like '%ped%') then 'Pedestrian'
+        when
+            lower(pcr_cause_of_injury) like '%motorcy%'
+            or lower(pcr_patient_complaints) like '%mc %'
+            then 'Motorcycle'
+        when lower(mvc_form_vehicle_type) = 'motorized scooter' then 'E-Scooter'
+        else 'Motor Vehicle'
+    end
+) stored;
+
+
+--
+-- add some indexes
+--
+create index ems__incidents_crashes_crash_pk_index on public.ems__incidents (
+    crash_pk
+);
+create index ems__incidents_crashes_incident_received_datetime_index
+on public.ems__incidents (incident_received_datetime);
+create index ems__incidents_apd_incident_numbers_index on ems__incidents using gin (
+    apd_incident_numbers
+);
+create index ems__incidents_geometry_index on ems__incidents using gist (
+    geometry
+);
+create index ems__incidents_patient_injry_sev_index on ems__incidents (
+    patient_injry_sev
+);
+create index ems__incidents_crash_match_status_index on ems__incidents (
+    crash_match_status
+);
+create index crashes_case_id_index on crashes (case_id);
+create index crashes_is_temp_record_index on crashes (
+    is_temp_record
+);
+-- these indexes optimize our ST_DWithin operations, which require geography types and are the fastes way to check point-to-point proximity
+create index idx_ems_geometry_geography on ems__incidents using gist (
+    (geometry::geography)
+);
+create index idx_crashes_position_geography on crashes using gist (
+    (position::geography)
+);
+
+--
+-- Create or replace the trigger function for the crashes table
+--
+create or replace function update_crash_ems_match()
+returns trigger as $$
+DECLARE
+    matching_ems RECORD;
+    ems_record RECORD;
+    match_count INTEGER;
+    matched_crash_ids INTEGER[];
+    meters_threshold INTEGER := 1000;
+    time_threshold INTERVAL := '60 minutes';
```

Comment:
I'm still working with Xavier to tune these to their optimal settings. We'll ultimately probably settle on tighter thresholds.

Reply:
yeah, fwiw the meter threshold feels far too liberal for me. when i was testing moving the crash so that it would unlink from the ems records, i felt like i had to move them so far away to finally unlink. feels too easy that the wrong crashes would get linked to ems records as is

Reply:
the thresholds we ultimately settled on are 1200 meters and a 30-min threshold (60-min window), which is even greedier! 

our thinking is that we will very likely adjust this after we-go live with the [UI updates](https://github.com/cityofaustin/atd-data-tech/issues/21852), which will make it easier for Xavier to start seeing how many false-positives we're creating, and how many "multiple matches" we're creating when the nearest crash was the obvious correct match.

the process of linking an EMS record to an actual _person_ record will be a manual process, so ultimately any data we use to inform official Vision Zero reporting will have go through manual QA.